File: LcWithPatterns/Python/Loops/while-loops/15-armstrong-number.py
n: int = 153
origional: int = n

temp: int = n
digits: int = 0

while temp > 0:
    digits += 1
    temp = temp // 10

sum_val: int = 0
current_n: int = n
while current_n > 0:
    digit: int = current_n % 10
    sum_val = sum_val + (digit ** digits)
    current_n = current_n // 10

if sum_val == origional:
    print(f"{origional} is an Armstrong number.")
else:
    print(f"{origional} is not an Armstrong number.")

File: LcWithPatterns/JavaScript/Loops/while-loops/15-armstrong-number.js
"use strict";
// Checking whether a number is an Armstrong number or not
Object.defineProperty(exports, "__esModule", { value: true });
let n = 153; // You can change this number to check for any other number
let origional = n; // Store the original number
let temp = n; // Temporary variable to calculate the number of digits
let digits = 0; // Initialize the count of digits to 0
// Calculate the total number of digits in the number
while (temp > 0) {
    digits++;
    temp = Math.floor(temp / 10); // Remove the last digit from the number
}
let sum = 0; // Initialize sum to 0
while (n > 0) {
    let digit = n % 10; // Get the last digit
    sum = sum + Math.pow(digit, digits); // Add the digit raised to the power of total digits to sum
    n = Math.floor(n / 10); // Remove the last digit from the number
}
if (sum === origional) {
    console.log(`${origional} is an Armstrong number.`);
}
else {
    console.log(`${origional} is not an Armstrong number.`);
}
